ui/app/lib/satellizer.js

```diff
@@ -0,0 +1,856 @@
+                var hash = utils.parseQueryString(hashParams);
+                var qs = utils.parseQueryString(queryParams);
+
+                angular.extend(qs, hash);
+
+                if (qs.error) {
+                  deferred.reject({ error: qs.error });
+                } else {
+                  deferred.resolve(qs);
+                }
+
+                popupWindow.close();
+                $interval.cancel(polling);
+              }
+            } catch (error) {
+            }
+
+            if (!popupWindow) {
+              $interval.cancel(polling);
+              deferred.reject({ data: 'Provider Popup Blocked' });
+            } else if (popupWindow.closed || popupWindow.closed === undefined) {
+              $interval.cancel(polling);
+              deferred.reject({ data: 'Authorization Failed' });
+            }
+          }, 35);
+          return deferred.promise;
+        };
+
+        popup.prepareOptions = function(options) {
+          var width = options.width || 500;
+          var height = options.height || 500;
+          return angular.extend({
+            width: width,
+            height: height,
+            left: $window.screenX + (($window.outerWidth - width) / 2),
+            top: $window.screenY + (($window.outerHeight - height) / 2.5)
+          }, options);
+        };
+
+        popup.stringifyOptions = function(options) {
+          var parts = [];
+          angular.forEach(options, function(value, key) {
+            parts.push(key + '=' + value);
+          });
+          return parts.join(',');
+        };
+
+        return popup;
+      }])
+    .service('satellizer.utils', function() {
+      this.camelCase = function(name) {
+        return name.replace(/([\:\-\_]+(.))/g, function(_, separator, letter, offset) {
+          return offset ? letter.toUpperCase() : letter;
+        });
+      };
+
+      this.parseQueryString = function(keyValue) {
+        var obj = {}, key, value;
+        angular.forEach((keyValue || '').split('&'), function(keyValue) {
+          if (keyValue) {
+            value = keyValue.split('=');
+            key = decodeURIComponent(value[0]);
+            obj[key] = angular.isDefined(value[1]) ? decodeURIComponent(value[1]) : true;
+          }
+        });
+        return obj;
+      };
+
+      this.joinUrl = function() {
+        var joined = Array.prototype.slice.call(arguments, 0).join('/');
+
+        var normalize = function(str) {
+          return str
+            .replace(/[\/]+/g, '/')
+            .replace(/\/\?/g, '?')
+            .replace(/\/\#/g, '#')
+            .replace(/\:\//g, '://');
+        };
+
+        return normalize(joined);
+      };
+    })
+    .factory('satellizer.storage', ['satellizer.config', function(config) {
+      var storage = function(type) {
+        var stub = {
+          getItem: function(key) { return undefined; },
+          setItem: function(key, value) { return undefined; },
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['value' is defined but never used.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2279397488/issues/source?bid=3144114&fileBranchId=3144125#l766)